tests/verify.rs

```diff
@@ -1,8 +1,170 @@
 use std::convert::TryFrom;
 
-use webpki::{EndEntityCert, KeyUsage, SubjectNameRef, Time};
+use rcgen::{
+    BasicConstraints, Certificate, CertificateParams, DnType, GeneralSubtree, IsCa,
+    KeyUsagePurpose, NameConstraints,
+};
+use webpki::{EndEntityCert, Error, KeyUsage, SubjectNameRef, Time};
+use x509_parser::extensions::{GeneralName, NameConstraints as X509ParserNameConstraints};
+use x509_parser::prelude::FromDer;
+
 use webpki_roots::TLS_SERVER_ROOTS;
 
+#[test]
+fn name_constraints() {
+    for name_constraints in TLS_SERVER_ROOTS.iter().filter_map(|ta| ta.name_constraints) {
+        let time = Time::from_seconds_since_unix_epoch(0x40000000); // Time matching rcgen default.
+        let test_case = name_constraints_testcase(name_constraints);
+        let trust_anchors =
```

Comment:
Maybe the `ConstraintTest` should contain the `Vec<TrustAnchor>` instead of the DER from rcgen?

Reply:
I think this will run into lifetime issues. The `TrustAnchor` is a reference type with a lifetime tied to the DER from rcgen that's local to the function where we create a `Certificate` and serialize it. We'd need an "owned" trust anchor like Rustls defines so it felt easier to pass back the owned `Vec<u8>`.

Reply:
Ah, fair enough.